api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -164,47 +165,99 @@ public void removeMember(Integer memberId) {
+	/**
+	 * @since 2.1.0
+	 */
+	public boolean removeMember(Patient patient) {
+		CohortMembership memberToRemove = this.getActiveMembership(patient);
+		if (memberToRemove != null && memberToRemove.getPatient().getPatientId().equals(patient.getPatientId())) {
+			memberToRemove.setEndDate(new Date());
+			return true;
+		}
+		return false;
 	}
-
-	public void addMembership(CohortMembership cohortMembership) {
-		if (!this.contains(cohortMembership.getPatient().getPatientId())) {
+	
+	/**
+	 * @since 2.1.0
+	 */
+	public boolean addMembership(CohortMembership cohortMembership) {
+		if (cohortMembership != null && !this.contains(cohortMembership.getPatient().getPatientId())) {
 			cohortMembership.setCohort(this);
-			getMembers().add(cohortMembership);
+			return getMemberships().add(cohortMembership);
 		}
+		return false;
 	}
 	
-	public void addMembership(Integer patientId) {
-		addMembership(new CohortMembership(new Patient(patientId)));
+	public boolean addMembership(Integer patientId) {
+		return addMembership(new CohortMembership(new Patient(patientId)));
 	}
 
+	/**
+	 * @since 2.1.0
+	 */
 	public void removeMembership(CohortMembership cohortMembership) {
-		List<CohortMembership> memberToRemoveList = getMembers().stream()
-				.filter(m -> m.equals(cohortMembership) ||
-						m.getPatient().getPatientId().equals(cohortMembership.getPatient().getPatientId()))
+		cohortMembership.setEndDate(new Date());
+	}
+	
+	/**
+	 * @since 2.1.0
+	 * @should return voided or unvoided memberships
+	 */
+	public List<CohortMembership> getMemberships(Boolean includeVoided) {
+		return getMemberships().stream()
+				.filter(m -> m.getVoided() == includeVoided)
 				.collect(Collectors.toList());
-		memberToRemoveList.forEach(m -> m.setEndDate(new Date()));
 	}
 
-	public List<CohortMembership> getMemberships(Date asOf) {
-		return getMembers().stream()
-				.filter(m -> !m.getStartDate().before(asOf)).collect(Collectors.toList());
+	/**
+	 * @since 2.1.0
+	 * @should get memberships of a cohort as of a date
+	 */
+	public List<CohortMembership> getActiveMemberships(Date asOfDate) {
+		return getMemberships().stream()
+				.filter(m -> m.isMembershipActive(asOfDate))
+				.collect(Collectors.toList());
+	}
+
+	public List<CohortMembership> getActiveMemberships() {
+		return getActiveMemberships(new Date());
+	}
+
+	/**
+	 * @since 2.1.0
+	 */
+	public Collection<CohortMembership> getMemberships() {
+		if (memberships == null) {
+			memberships = new TreeSet<CohortMembership>();
+		}
+		return memberships;
 	}
 
+	/**
+	 * @since 2.1.0
+	 */
+	public CohortMembership getActiveMembership(Patient patient) {
+		return getMemberships().stream()
+				.filter(m -> m.isMembershipActive() && m.getPatient().getPatientId().equals(patient.getPatientId()))
+				.collect(Collectors.toList())
+				.get(0);
+	}
+	
+	/**
+	 * @since 2.1.0
+	 */
 	public void purgeMemberships(List<CohortMembership> cohortMembershipList) {
```

Comment:
I would also assume this should return a boolean value, i.e only return true if the member was found and removed by return the same value  from `getMemberships().remove(m)`